### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: backend/remote-state/http/client.go [hunk 1]
```diff
@@ -97,7 +97,6 @@ func (c *httpClient) Lock(info *state.LockInfo) (string, error) {
 	case http.StatusForbidden:
 		return "", fmt.Errorf("HTTP remote state endpoint invalid auth")
 	case http.StatusConflict, http.StatusLocked:
-		defer resp.Body.Close()
```
Comment (ash2k) [inline backend/remote-state/http/client.go:100]: This has been done already on line 88 above.

**Comment by mildwonkey** [comment]
Thasnk @ash2k ! Can you please split this into two PRs, one with the typo fixes and the other with the change to `backend/remote-state/http/client.go` ? It's easiest for us to review and merge pull requests that don't have mixed purposes - many of the backends are maintained by different teams or individuals.

**Comment by ash2k** [comment]
@mildwonkey done. HTTP backend changes are in #24639.

**Comment by mildwonkey** [comment]
Thanks so much!

**Pull request merged**